templates/addons/aurora/mysql-cf.yml

```
@@ -0,0 +1,171 @@
+Parameters:
+  App:
+    Type: String
+    Description: Your application's name.
+  Env:
+    Type: String
+    Description: The environment name your service, job, or workflow is being deployed to.
+  Name:
+    Type: String
+    Description: The name of the service, job, or workflow being deployed.
+  # Customize your Aurora Serverless cluster by setting the default value of the following parameters.
+  InitialDBName:
+    Type: String
+    Description: The name of the table.
+    Default: {{.InitialDBName}}
+    # Cannot have special characters
+    # Naming constraints: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Limits.html#RDS_Limits.Constraints
+  DBPort:
+    Type: Number
+    Description: The traffic port for connections to the database.
+    Default: 3306 # MySQL
+    # Default: 5432 # PostgreSQL
```

Can we inject this differently based on the engine? 
```suggestion
{[- if eq .Engine "aurora-postgresql"}}
    Default: 5432
{{- else}}
    Default: 3306
{{- end}}
```

---

Alternaatively, since I see that there's a file for postgres down below, can we remove "port" from the params entirely?

---

Ya I initially just have one single metatemplate "cf.yml" file that contains a bunch of `{{if}}{{else}}{{end}}` statements. Then I thought since the logic is all over the place I might as well use different meta-template for different engine.

Do we have any preferences between the two approaches?

---

I think it’s better to have two separate templates since the default values for the parameter groups will also have to change. Let’s keep things separate and remove any references to PostgreSQL in comments on the MySQl template and vice versa.

Let’s also remove `DBPort` as a parameter and just hard-code it in the templates.

---

Should we keep the PostgreSQL comment so that it's easier for the customers to make the change in case they want to switch from MySQL to using PostgreSQL (and vice versa)? They would just need to un-comment those lines.

---

It's not as simple as just changing the engine version in a variable. The parameter group also has to change, and when we add rotation, the hosted rotation lambda has to be totally updated. i think it's better to encourage folks to generate a new storage addon by re-running the command; otherwise we're just asking to support a whole bunch of tangled, difficult to troubleshoot CF.

---

Yea that makes sense. Will update! Thanks for explaining!